presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/ColumnMapping.java

```diff
@@ -0,0 +1,104 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc;
+
+import io.prestosql.spi.predicate.Domain;
+import io.prestosql.spi.type.Type;
+
+import java.util.function.Predicate;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public final class ColumnMapping
+{
+    public static ColumnMapping booleanMapping(Type prestoType, BooleanReadFunction readFunction, BooleanWriteFunction writeFunction)
+    {
+        return new ColumnMapping(prestoType, readFunction, writeFunction, domain -> true);
+    }
+
+    public static ColumnMapping longMapping(Type prestoType, LongReadFunction readFunction, LongWriteFunction writeFunction)
+    {
+        return new ColumnMapping(prestoType, readFunction, writeFunction, domain -> true);
+    }
+
+    public static ColumnMapping doubleMapping(Type prestoType, DoubleReadFunction readFunction, DoubleWriteFunction writeFunction)
+    {
+        return new ColumnMapping(prestoType, readFunction, writeFunction, domain -> true);
+    }
+
+    public static ColumnMapping sliceMapping(Type prestoType, SliceReadFunction readFunction, SliceWriteFunction writeFunction)
+    {
+        return new ColumnMapping(prestoType, readFunction, writeFunction, domain -> true);
+    }
+
+    private final Type type;
+    private final ReadFunction readFunction;
+    private final WriteFunction writeFunction;
+    private final Predicate<Domain> predicatePushdownFilter;
+
+    private ColumnMapping(Type type, ReadFunction readFunction, WriteFunction writeFunction, Predicate<Domain> predicatePushdownFilter)
+    {
+        this.type = requireNonNull(type, "type is null");
+        this.readFunction = requireNonNull(readFunction, "readFunction is null");
+        this.writeFunction = requireNonNull(writeFunction, "writeFunction is null");
+        checkArgument(
+                type.getJavaType() == readFunction.getJavaType(),
+                "Presto type %s is not compatible with read function %s returning %s",
+                type,
+                readFunction,
+                readFunction.getJavaType());
+        checkArgument(
+                type.getJavaType() == writeFunction.getJavaType(),
+                "Presto type %s is not compatible with write function %s accepting %s",
+                type,
+                writeFunction,
+                writeFunction.getJavaType());
+        this.predicatePushdownFilter = requireNonNull(predicatePushdownFilter, "predicatePushdownFilter is null");
+    }
+
+    public Type getType()
+    {
+        return type;
+    }
+
+    public ReadFunction getReadFunction()
+    {
+        return readFunction;
+    }
+
+    public WriteFunction getWriteFunction()
+    {
+        return writeFunction;
+    }
+
+    public Predicate<Domain> getPredicatePushdownFilter()
```

Comment:
Maybe let's change it to `Function<Domain, Domain> getDomainConverter()` so that it's possible to simplify domain if they are too large for JDBC database (e.g: Oracle)

Reply:
Please see https://github.com/prestosql/presto/pull/109/commits/f61a7f0ab69aca8481b7e46d6c863ab896a5b20b